create wget-task version1 (#357)
tektoncd/catalog: wget/README.md

# `wget`

> This task performs wget operation to download file from internet to workspace .

Reviewer: ```suggestion
This task uses wget  to download files from the internet to a workspace .
```

Author: Thanks, Dibyo, now the description makes more sense!

## Install the Task

```
kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/v1beta1/wget/wget.yaml
```

## Parameters

> * **mwendler/wget**: `wget` CLI container image to run this task.

Reviewer: Suggest we rename it to `wget-image` or something that makes it clear that this is the container image to be used. Also we should mention the default value for this paramter

Reviewer: Also, looks like this is not actually a parameter that is used? I think it is ok for this to not be a param. The `wget` command does not change frequently so its less likely that someone might want to use a different image

Author: Ok, I will just delete this line

* **options**: The arguments to pass to `wget` CLI.  _default_: `[""]`
* **diroptions**: The directory path we want to save file to.  _default_: `[""]`
* **url**: The url we want to download file from.  _default_: `""`
* **filename**: The filename we want to change our file to.  _default_: `""`
## Usage

### Running the Task

#### Download file to the workspace

You can invoke `wget url -P workspace` to check if file is downloaded to workspace:

Create a file, `wget-run.yaml`:

```
apiVersion: tekton.dev/v1beta1
kind: TaskRun
metadata:
  name: wget-run
spec:
  params:
    - name: url
      value: "www.google.com"
    - name: diroptions
      value:
        - "-P"
  taskRef:
    name: wget
  workspaces:
    - name: myworkspace
      emptyDir: {}
```

> Run it with `kubectl create -f wget-run.yaml`

Reviewer: nit: the taskrun uses name but the kubectl command usage uses create. We should either switch the taskrun to use `generateName` or use `kubectl apply`
